Re-Former

Div685 opened the pull request:
This project was about creating a simple HTML Form Using Rails.

**In this milestone we:**
- Added a User model 
- Migrated database
- Used Bootstrap to style the Form

**Features**
- Add New Users
- Edit Users

cyonii committed:
Add User model

- validate and migrate user model
- create #new and #create controller actions

Div685 committed:
Add HTML User Form

- Add create and user_params method

cyonii committed:
Rebuild form using #form_tag

cyonii committed:
Rebuild form using #form_for

Div685 committed:
Add validations and Edit html

- Add template partial file
- Add validation

cyonii committed:
Add Github workflow actions and rubocop config files

cyonii committed:
Style views with bootstrap

Div685 committed:
update readme.md file

- Add Screenshot image

Div685 committed:
update readme.md file

- Removed live demo link

Div685 committed:
update file for linter

SafaErden commented on README.md:
```diff
@@ -2,3 +2,87 @@
+
+<!-- This project is based on the . -->
+
+This is project about the creating the simple HTML form Using Rails. This project is based on The Odin Project’s Ruby on Rails Curriculum. Find it at http://www.theodinproject.com.
```
- [x] Your README contains some typos and grammar mistakes, please kindly fix them(dependency, This is project about the creating the simple ...,)

SafaErden commented on app/models/user.rb:
```diff
@@ -0,0 +1,13 @@
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    pattern = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+    record.errors[attribute] << (options[:message] || 'is not an email') unless value =~ pattern
+  end
+end
+
+class User < ApplicationRecord
```
- [x] If you are grouping classes with related functionality into a single file then you should also consider making them part of the same module for a namespace. So it would be better to create a helper function to validate email instead of creating another class.

SafaErden reviewed (changes requested):
## Status: Require changes ♻️

**Hi TEAM, your project is awesome. 👐 However, we need a little fix before we can proceed. Kindly find comments below ⤵**
##

**REQUIRED CHANGES**

- [x] For your PR message, please consider the tense consistency(past tense or simple present tense)

- [x] Your commented out `form` is missing with the features below:
- uses `users_path` helper for action
- uses `post` for method
- uses `"UTF-8"` for ccept-charset
- uses `user[<field_name>]` for input's name
- uses form_authenticity_token
Please kindly add it. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

- [x] Your commented out `form_tag` is missing with the features below:
- uses `label_tag`, `text_field_tag`, `email_field_tag`, `password_field_tag`
Please kindly add it. 
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

- [x] Your form built with `form_for` should use `@user` from controller
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_for.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

##

**OPTIONAL CHANGES**

- [x] Since you check the length of the password, you don't need to add presence validation, but it would be better if you validate the presence of it.(it has validations for the presence of each of the fields ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_model_attr_validation.png) from Odin))

##

**NOTES**

Friendly reminder: Remember it's important to have a professional repo so you can increase your job opportunities. Please take a step back and read the lesson [here.](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/)

##

**SUMMARY**

As I mentioned above, your project is almost done. You have some minor issues to solve. Please be an optimist and do not forget that the main purpose is to learn and practice. Do not let the requested changes discourage you 👍

##

*Kindly make the changes and submit another code review when you are ready💪*

Questions or comments? Reach me on [Twitter](https://twitter.com/safaerden) or [LinkedIn](https://www.linkedin.com/in/safaerden/) 🗨️

Happy coding_ 👨‍💻 

**Safa ERDEN**

Div685 committed:
update files as requested

- Add form comment in both new and edit.html form
- removied Email validation class from user model
- Add form_tag comment
- update Readme.md file

lekegitrepo reviewed (changes requested):
## Status: Changes Required! :recycle:
### Hi @Div685 and @cyonii :clap: :clap:
#### You have done a good implementing the required changes from the previous review. However, you need to fix the following :point_down: issues before this project can be approved.

#### Missing Required Changes(from the previous review):
- _This [`accept-charset="utf-8"`] should be added to the commented `form`_
-  _Your form built with `form_for` should use `@user` from the controller_

#### App Behaviour:
- _When I tried to load the `users/new` page it outputted this error:_

![Screenshot from 2020-12-16 05-17-36](https://user-images.githubusercontent.com/48312803/102342952-976edc00-3f5f-11eb-8614-7c14905f9620.png)

#### _Reviewed by Temitayo Adeleke_

Div685 committed:
update user model

- Add UTF-8 to commented form

andresporras3423 reviewed (approved):
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:
**Hello Cy Kalu & Divyesh Patel** 
Amazing job :tada: :tada: :tada:  you have met all the requirements for this project
Now you are ready to continue with your next challenge :muscle: 
Don't forget to merge with the develop branch
 
Keep going like this  :clap: :clap:
 
Oscar Andrés Russi Porras
